pandas/core/indexes/base.py

```diff
@@ -7478,7 +7480,14 @@ def ensure_index_from_sequences(sequences, names=None) -> Index:
     if len(sequences) == 1:
         if names is not None:
             names = names[0]
-        return Index(maybe_sequence_to_range(sequences[0]), name=names)
+        data = sequences[0]
+        conv_data = maybe_sequence_to_range(data)
+        dtype = (
+            data.dtype
+            if isinstance(data, np.ndarray) and isinstance(conv_data, range)
+            else None
+        )
+        return Index(conv_data, dtype=dtype, name=names)
```

Comment:
https://github.com/pandas-dev/pandas/issues/59077#issuecomment-2241171136
I think this is the best place to add in the explicit dtype after the possible conversion to python native `range` type.